landscape/client/manager/tests/test_shutdownmanager.py

```diff
@@ -18,179 +12,38 @@ class ShutdownManagerTest(LandscapeTest):
 
-    def test_restart(self):
-        """
-        C{shutdown} processes run until the shutdown is to be performed.  The
-        L{ShutdownProcessProtocol} watches a process for errors, for 10
-        seconds by default, and if none occur the activity is marked as
-        L{SUCCEEDED}.  Data printed by the process is included in the
-        activity's result text.
-        """
-        message = {"type": "shutdown", "reboot": True, "operation-id": 100}
-        self.plugin.perform_shutdown(message)
-        [arguments] = self.process_factory.spawns
-        protocol = arguments[0]
-        self.assertTrue(isinstance(protocol, ShutdownProcessProtocol))
-        self.assertEqual(
-            arguments[1:3],
-            (
-                "/sbin/shutdown",
-                [
-                    "/sbin/shutdown",
-                    "-r",
-                    "+4",
-                    "Landscape is rebooting the system",
-                ],
-            ),
-        )
+        self.clock = task.Clock()
+        self.plugin = ShutdownManager()
+        self.plugin.reactor = self.clock
 
-        def restart_performed(ignore):
-            self.assertTrue(self.broker_service.exchanger.is_urgent())
-            self.assertEqual(
-                self.broker_service.message_store.get_pending_messages(),
-                [
-                    {
-                        "type": "operation-result",
-                        "api": b"3.2",
-                        "operation-id": 100,
-                        "timestamp": 10,
-                        "status": SUCCEEDED,
-                        "result-text": "Data may arrive in batches.",
-                    },
-                ],
-            )
-
-        protocol.result.addCallback(restart_performed)
-        protocol.childDataReceived(0, b"Data may arrive ")
-        protocol.childDataReceived(0, b"in batches.")
-        # We need to advance both reactors to simulate that fact they
-        # are loosely in sync with each other
-        self.broker_service.reactor.advance(10)
-        self.manager.reactor.advance(10)
-        return protocol.result
+        self.manager.add(self.plugin)
 
-    def test_shutdown(self):
-        """
-        C{shutdown} messages have a flag that indicates whether a reboot or
-        shutdown has been requested.  The C{shutdown} command is called
-        appropriately.
-        """
-        message = {"type": "shutdown", "reboot": False, "operation-id": 100}
-        self.plugin.perform_shutdown(message)
-        [arguments] = self.process_factory.spawns
-        self.assertEqual(
-            arguments[1:3],
-            (
-                "/sbin/shutdown",
-                [
-                    "/sbin/shutdown",
-                    "-h",
-                    "+4",
-                    "Landscape is shutting down the system",
-                ],
-            ),
-        )
+    def tearDown(self):
+        return super().tearDown()
```

Comment:
If you just call `super` then there's no point in overriding `tearDown` and you can just delete it.

Reply:
OK - thanks and noted.